File: packages-gen/js/db/system/main-client.js
const js_sqlite = require("@xtalk/db/net/conn-sqlite.js")

const js_ws = require("@xtalk/db/net/ws-native.js")

const js_postgres = require("@xtalk/db/net/conn-postgres.js")

const addon = require("@xtalk/net/addon-supabase.js")

const js_fetch = require("@xtalk/db/net/http-fetch.js")

function create_client(type,defaults){
  if(type == "sqlite"){
    return js_sqlite.create(defaults);
  }
  else if(type == "postgres"){
    return js_postgres.create(defaults);
  }
  else if(type == "supabase"){
    let client = js_fetch.create(defaults,addon.middleware_supabase());
    client["create_ws_client"] = js_ws.create;
    return client;
  }
  else{
    return null;
  }
}

module.exports = {["create_client"]:create_client}

File: packages-gen/dart/db/lib/system/main-client.dart
import 'xtalk_db/net/http-fetch.dart' as dart_fetch;

import 'xtalk_db/net/ws-native.dart' as dart_ws;

import 'xtalk_db/net/conn-postgres.dart' as dart_postgres;

import 'package:xtalk_net/addon-supabase.dart' as addon;

import 'xtalk_db/net/conn-sqlite.dart' as dart_sqlite;

create_client(type, defaults) {
  if(type == "sqlite"){
    return dart_sqlite.create(defaults);
  }
  else if(type == "postgres"){
    return dart_postgres.create(defaults);
  }
  else if(type == "supabase"){
    var client = dart_fetch.create(defaults,addon.middleware_supabase());
    client["create_ws_client"] = ((ws_defaults) {
      return dart_ws.create(
        xt.lang.common_data.obj_assign(ws_defaults,<dynamic, dynamic>{"background":true})
      );
    });
    return client;
  }
  else{
    return null;
  }
}